src/mlpack/bindings/python/mlpack/matrix_utils.py

```diff
@@ -41,129 +41,142 @@ def to_matrix(x, dtype=np.double, copy=False):
   """
   Given some array-like X, return a numpy ndarray of the same type.
   """
-  # Make sure it's array-like at all.
-  if not hasattr(x, '__len__') and \
-      not hasattr(x, 'shape') and \
-      not hasattr(x, '__array__'):
-    raise TypeError("given argument is not array-like")
-
-  if (isinstance(x, np.ndarray) and x.dtype == dtype and x.flags.c_contiguous):
-    if copy: # Copy the matrix if required.
-      return x.copy("C"), True
-    else:
-      return x, False
-  elif (isinstance(x, np.ndarray) and x.dtype == dtype and x.flags.f_contiguous):
-    if copy: # Copy the matrix if required.
-      return np.ndarray(x.shape, buffer=x.flatten(), dtype=dtype, order='C').copy("C"), True
-    else:
-      return np.ndarray(x.shape, buffer=x.flatten(), dtype=dtype, order='C'), False
-  else:
-    if isinstance(x, pd.core.series.Series) or isinstance(x, pd.DataFrame):
-      y = x.values
+  # Check the Type of input X
+  if isinstance(x, np.ndarray) or isinstance(x, pd.DataFrame) or isinstance(x,
+      pd.Series) or isinstance(x, list) or isinstance(x, tuple):
+
+    # Make sure it's array-like at all.
+    if not hasattr(x, '__len__') and \
+        not hasattr(x, 'shape') and \
+        not hasattr(x, '__array__'):
+      raise TypeError("given argument is not array-like")
```

Comment:
Ah, sorry, I meant that the check is already in these functions, and it's this one right here.  So I think we can leave `to_matrix()` and `to_matrix_with_info()` as they were before the changes you made, since they already throw `TypeError`s if they are not array-like.  (A nice side-effect of this way of checking, just for `__len__`, `shape`, and `__array__`, means that other array-like things can also be passed to Python bindings, not just numpy and Pandas matrices.)